### ISSUE TITLE
Deprecate DirectLine.start()

### ISSUE THREAD

**Issue by billba** [Microsoft/BotFramework-WebChat#229]
I’d like to make a subtle change to the DirectLine lifecycle.
 
A while back I added a `start()` method to DirectLine in order to distinguish between creating a DirectLine object (using the constructor) and opening an actual connection to the bot. I also added a corresponding `end()` method to unsubscribe to the various things in DirectLine subscribes too.
 
This turns out to cause a problem with backchannel. The way backchannel works is that the web page creates a DirectLine object which it passes to Chat but also keeps a hold of to send and/or receive messages with the bot. The problem is that, right now, the web page needs to wait until the Chat calls start() by checking `connectionStatus$`. This works, but it's clunky.
 
On analysis, we don't really need `start()` at all. Instead, DirectLIne can just open the connection (start the conversation) when any of the following happen:
·         Someone subscribes to activity$
·         Someone calls postActivity
·         Someone calls postMessageWithAttachments
 
In other words, the connection is made when it’s needed. Again, the current state can be determined by checking `connectionStatus$`.

I would keep a non-functional version of start() around for a little while, so as give @eanders-MS time to update the emulator by removing this call.

**Comment by eanders-MS** [comment]
Sounds fine to me @billba, but no need to keep a stubbed `start()` for my benefit. It should be trivial to adapt the emulator.
